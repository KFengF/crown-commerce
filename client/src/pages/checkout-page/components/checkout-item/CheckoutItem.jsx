import React from 'react';
import { connect } from 'react-redux';
import { removeItem, addItem, decreaseItem } from '../../../../utils/redux/cart/cartActions';
import { CheckoutItemDiv, ImageDiv, NameSpan, QuantityDiv, PriceSpan, RemoveSpan } from './CheckoutItemStyles';
/* import './CheckoutItem.scss'; */

const CheckoutItem = ({ cartItem, removeItem, decreaseItem, addItem }) => {
    const { name, imageUrl, price, quantity } = cartItem;
    return (
        <CheckoutItemDiv>
            <ImageDiv>
                <img src={ imageUrl } alt="checkout item" />
            </ImageDiv>
            <NameSpan>{ name }</NameSpan>
            <QuantityDiv >
                <span
                    onClick={ () => decreaseItem(cartItem) }
                    style={{ marginLeft: 'auto' }}
                >
                    &#10094;
                </span>
                <p>{ quantity }</p>
                <span
                    onClick={ () => addItem(cartItem) }
                    style={{ marginRight: 'auto' }}
                >
                    &#10095;
                </span>
            </QuantityDiv>
            <PriceSpan>{ price }</PriceSpan>
            <RemoveSpan onClick={ () => removeItem(cartItem) } >
                &#10005;
            </RemoveSpan>
        </CheckoutItemDiv>
    );
}

const mapDispatchToProps = dispatch => ({
    removeItem: item => dispatch(removeItem(item)),
    addItem: item => dispatch(addItem(item)),
    decreaseItem: item => dispatch(decreaseItem(item))
});

export default connect(null, mapDispatchToProps)(CheckoutItem);