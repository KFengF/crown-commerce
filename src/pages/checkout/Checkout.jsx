import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import CheckoutItem from './components/checkout-item/CheckoutItem';
import { cartItemsSelector, cartItemsTotalSelector } from '../../utils/redux/cart/cartSelector';
import './Checkout.scss';

const Checkout = ({ cartItems, cartItemsTotal }) => (
    <div className="checkout">
        <div className="checkout-header">
            <div className="header-block">
                <span>Product</span>
            </div>
            <div className="header-block">
                <span>Description</span>
            </div>
            <div className="header-block">
                <span>Quantity</span>
            </div>
            <div className="header-block">
                <span>Price</span>
            </div>
            <div className="header-block">
                <span>Remove</span>
            </div>
        </div>
        { cartItems.map(cartItem => 
            <CheckoutItem key={ cartItem.id } cartItem={ cartItem } />
        ) }
        <span className="total">TOTAL: ${ cartItemsTotal }</span>
    </div>
);

const mapStateToProps = createStructuredSelector({
    cartItems: cartItemsSelector,
    cartItemsTotal: cartItemsTotalSelector
});

export default connect(mapStateToProps)(Checkout);