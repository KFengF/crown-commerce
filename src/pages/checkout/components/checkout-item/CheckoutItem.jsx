import React from 'react';
import { connect } from 'react-redux';
import { removeItem, addItem, decreaseItem } from '../../../../utils/redux/cart/cartActions';
import './CheckoutItem.scss';

const CheckoutItem = ({ cartItem, removeItem, decreaseItem, addItem }) => {
    const { name, imageUrl, price, quantity } = cartItem;
    return (
        <div className="checkout-item">
            <div className="image-container">
                <img src={ imageUrl } alt="checkout item" />
            </div>
            <span className="name" >{ name }</span>
            <div className="quantity" >
                <div 
                    className="arrow" 
                    onClick={ () => decreaseItem(cartItem) }
                >
                    &#10094;
                </div>
                <span className="value">{ quantity }</span>
                <p className="arrow" onClick={ () => addItem(cartItem) } >&#10095;</p>
            </div>
            <span className="price" >{ price }</span>
            <div 
                className="remove-button" 
                onClick={ () => removeItem(cartItem) }
            >
                &#10005;
            </div>
        </div>
    );
}

const mapDispatchToProps = dispatch => ({
    removeItem: item => dispatch(removeItem(item)),
    addItem: item => dispatch(addItem(item)),
    decreaseItem: item => dispatch(decreaseItem(item))
});

export default connect(null, mapDispatchToProps)(CheckoutItem);