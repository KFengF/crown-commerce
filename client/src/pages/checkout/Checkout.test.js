import React from 'react';
import { shallow } from 'enzyme';
import { CheckoutPage } from './Checkout.page';
import CheckoutItem from './components/checkout-item/CheckoutItem.component';

describe('Checkout page component', () => {
    let mockProps;
    let wrapper;
    const mockCartItems = [{
        id: 12,
        name: 'a',
        imageUrl: 'https://example.com/img?img=1',
        price: 10,
        quantity: 1
    }, {
        id: 13,
        name: 'b',
        imageUrl: 'https://example.com/img?img=2',
        price: 20,
        quantity: 2
    }, {
        id: 14,
        name: 'c',
        imageUrl: 'https://example.com/img?img=3',
        price: 13,
        quantity: 3
    }];

    beforeEach(() => {
        mockProps = {
            cartItems: mockCartItems,
            cartItemsTotal: 43
        }

        wrapper = shallow(<CheckoutPage { ...mockProps } />);
    })

    it('should render Checkout page', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render an equal number of CheckoutItem components as the cartItems prop', () => {
        expect(wrapper.find(CheckoutItem).length).toEqual(mockCartItems.length);
    });

    it('should be the text TOTAL: $43 in TotalSpan', () => {
        expect(wrapper.find('TotalSpan').text()).toBe('TOTAL: $43');
    });
});