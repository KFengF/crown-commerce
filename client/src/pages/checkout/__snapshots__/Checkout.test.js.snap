// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkout page component should render Checkout page 1`] = `ShallowWrapper {}`;
